### PR TITLE
Sctfen

### PR DESCRIPTION
**Pull request by sctfn**
I've not merged back in with develop since 2015, so here's what I've done off and on since then.

### CODE REVIEW

**Comment by sctfn** [comment]
NOTE: failures are actually b/c I've reduced OLD usage and pruned some old ones in my mathbox.

**Comment by david-a-wheeler** [comment]
Wow, this seems to involve a lot more than set.mm.

It changes 130 files, including
* other-dbs/abgp-eq.mm
* other-dbs/axesonly.mm
*  other-dbs/co.mm
* scripts/... including scripts/mmcheck.py
* Java files

This would be a big change in scope - currently the "set.mm" project only covers the "set.mm" database.  It seems to me that we want to keep the different databases separate, and the databases separate from the programs used to process them.

**Comment by digama0** [comment]
I noticed your branch earlier, and my reaction is much the same as David's. This is a whole different project you are working on, which just happens to involve some additions to set.mm as an aside. I would recommend that you put all this in a separate github repo of your own, and just commit the changes to set.mm here. I don't think it likely that any of the other stuff will be making it on this repo, which is supposed to be primarily for set.mm, without a long discussion.

**Comment by david-a-wheeler** [comment]
There *are* some changes to set.mm here, within Scott Fenton's mathbox.

I suggest that you create a separate pull request for *just* the changes to set.mm.  I can imagine some of those changes even moving from the mathbox into the body.  The subtraction of reciprocals  ` ( ( 1 / A ) - ( 1 / B ) ) = ( ( B - A ) / ( A x. B ) ) ` looks generic and useful.  While more specialized, specific ones like this: ` ( ( 1 / 2 ) - ( 1 / 3 ) ) = ( 1 / 6 ) ` are clearly true for the main set.mm and who knows, they might eventually go into the main body too.

**Comment by digama0** [comment]
Hm, that theorem seems very similar to halfpm6th: 

    |- ( ( ( 1 / 2 ) - ( 1 / 6 ) ) = ( 1 / 3 ) /\ ( ( 1 / 2 ) + ( 1 / 6 ) ) = ( 2 / 3 ) )

**Comment by david-a-wheeler** [comment]
@digama0 - sure.  I was just commenting that adding these to a set.mm mathbox is perfectly reasonable.

**Comment by sctfn** [comment]
OK, let me modify my repo here to just have the set.mm stuff. I'll recreate
the pull req afterwards.

**Comment by sctfn** [comment]
OK, I've segregated out my personal stuff. Please pull from here.

**Comment by david-a-wheeler** [comment]
Awesome, thanks!  I see there are some tabs, though.  The metamath program supports them, but stylistically I believe we always use spaces.  Do you mind changing them?

This looks like an unspoken rule, sorry about that.  I'll propose adding note about tabs in "conventions" and creating a runtime check for this.

**Comment by sctfn** [comment]
Done and done. Thanks!

On Tue, Jan 10, 2017 at 10:58 AM, David A. Wheeler <notifications@github.com

File: set.mm [hunk 1]
```diff
@@ -1,4 +1,5 @@
-$( set.mm - Version of 7-Jan-2017 $)
+$( -*-text-*- $)
```
Comment (digama0) [inline set.mm]: What is this line? Could you remove it?

**Comment by sctfn** [comment]
Ach, my bad. I edit in emacs, and that line puts emacs in text mode. I'll
remove it.

**Comment by sctfn** [comment]
Done.

**Comment by digama0** [comment]
Do you see a problem with a squash merge? There is a lot of mess in the history here (in particular those extra files), which I can clean up by squashing everything into a single commit. Other than that, I think this is good to go.

**Comment by nmegill** [comment]
'save proof */compressed/fast' together with 'write source 
set.mm/rewrap' should convert all tabs to spaces, under the assumption 
of uniform tab stops every 8 columns. (If tabs aren't being converted, 
let me know and I should fix that - perhaps I overlooked a case because 
my editor also converts them.)

So I don't think it is an issue worth documenting since they will be 
converted anyway.  If people prefer tabs while doing their work it's ok 
with me,

The reason for preferring spaces over tabs is to make it look consistent 
for everyone.  I don't know if people still set custom tab stops in 
their editors, but years ago, before /rewrap, this was an occasional 
problem.  Although tabs used to serve as an inexpensive way to compress 
spaces, I don't think that's much of an issue anymore.

Norm

**Comment by sctfn** [comment]
Nah, no problem with the squash merge. Thanks for walking me through all of
this.

**Pull request merged**